File: bugs/core/world/entities/ant/base/genome/genome.py
from .chromosomes_set import ChromosomesSet

class Genome():

    @classmethod
    def build(cls, maternal_chromosomes_set: ChromosomesSet, paternal_chromosomes_set: ChromosomesSet):
        return Genome(maternal_chromosomes_set, paternal_chromosomes_set)

    def __init__(self, maternal_chromosomes_set: ChromosomesSet, paternal_chromosomes_set: ChromosomesSet):
        self._maternal_chromosomes_set = maternal_chromosomes_set
        self._paternal_chromosomes_set = paternal_chromosomes_set

    @property
    def maternal_chromosomes_set(self) -> ChromosomesSet:
        return self._maternal_chromosomes_set
    
    @property
    def paternal_chromosomes_set(self) -> ChromosomesSet:
        return self._paternal_chromosomes_set

File: bugs/core/client/app/src/domain/entity/genetic/genome.js
import { ChromosomesSet } from "./chromosomesSet";

class Genome {

    static buildFromJson(json) {
        let maternalChromosomesSet = new ChromosomesSet(json.maternal);
        let paternalChromosomesSet = json.paternal ? new ChromosomesSet(json.paternal) : null;
        return new Genome(maternalChromosomesSet, paternalChromosomesSet, json.avaliableAntTypes);
    }

    constructor(maternal, paternal, avaliableAntTypes) {
        this._maternal = maternal;
        this._paternal = paternal;
        this._avaliableAntTypes = avaliableAntTypes;
    }
    
    get maternal() {
        return this._maternal;
    }
    
    get paternal() {
        return this._paternal;
    }
    
    get avaliableAntTypes() {
        return this._avaliableAntTypes;
    }

    get isFertilized() {
        return !!this._paternal;
    }

}

export {
    Genome
}